src/plugins/ui_feature/selection_input.ts

```diff
@@ -185,7 +185,7 @@ export class SelectionInputPlugin extends UIPlugin implements StreamCallbacks<Se
       ...values.map((xc, i) => ({
         xc,
         id: (this.ranges.length + i + 1).toString(),
-        color: getNextColor(),
+        color: getNextColor(this.ranges.length + i),
```

Comment:
The helper does not make sense anymore if you have to provide an index to a function called "getNextColor", it defeats the purpose. 
```suggestion
        color: colors[this.ranges.length +1 % colors.length]
```
should do the trick and also address the other issue I  raise on the other comment

Reply:
Was not sure if we needed the helper function in any future case that's why I didn't removed it earlier but now I removed it and added your suggested change